File: app/src/main/java/com/bryan/platform/model/entity/Moment.java
package com.bryan.platform.model.entity;

import androidx.annotation.NonNull;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;
import java.util.Collections;
import java.util.List;

/**
 * 动态(朋友圈)实体数据模型（Android 客户端用）
 * 对应后端的 Moment MongoDB 文档
 */
public class Moment implements Serializable {

    @SerializedName("id")
    private String id;

    @SerializedName("content")
    private String content;

    @SerializedName("images")
    private String images;          // JSON 字符串，客户端按需解析

    @SerializedName("authorId")
    private Long authorId;

    @SerializedName("authorName")
    private String authorName;

    @SerializedName("likeCount")
    private Integer likeCount = 0;

    @SerializedName("comments")
    private List<Comment> comments = Collections.emptyList();

    // 注意：后端字段为 snake_case
    @SerializedName("created_at")
    private String createdAt;

    @SerializedName("updated_at")
    private String updatedAt;

    public Moment() {
    }

    public Moment(String id,
    String content,
    String images,
    Long authorId,
    String authorName,
    Integer likeCount,
    List<Comment> comments,
    String createdAt,
    String updatedAt) {
        this.id = id;
        this.content = content;
        this.images = images;
        this.authorId = authorId;
        this.authorName = authorName;
        this.likeCount = likeCount;
        this.comments = comments;
        this.createdAt = createdAt;
        this.updatedAt = updatedAt;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public String getImages() {
        return images;
    }

    public void setImages(String images) {
        this.images = images;
    }

    public Long getAuthorId() {
        return authorId;
    }

    public void setAuthorId(Long authorId) {
        this.authorId = authorId;
    }

    public String getAuthorName() {
        return authorName;
    }

    public void setAuthorName(String authorName) {
        this.authorName = authorName;
    }

    public Integer getLikeCount() {
        return likeCount;
    }

    public void setLikeCount(Integer likeCount) {
        this.likeCount = likeCount;
    }

    public List<Comment> getComments() {
        return comments;
    }

    public void setComments(List<Comment> comments) {
        this.comments = comments;
    }

    public String getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(String createdAt) {
        this.createdAt = createdAt;
    }

    public String getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(String updatedAt) {
        this.updatedAt = updatedAt;
    }

    @NonNull
    @Override
    public String toString() {
        return "Moment{" +
                "id='" + id + '\'' +
                ", content='" + content + '\'' +
                ", images='" + images + '\'' +
                ", authorId=" + authorId +
                ", authorName='" + authorName + '\'' +
                ", likeCount=" + likeCount +
                ", comments=" + comments +
                ", createdAt='" + createdAt + '\'' +
                ", updatedAt='" + updatedAt + '\'' +
                '}';
    }
}

File: app/src/main/java/com/bryan/platform/model/entity/Moment.kt
package com.bryan.platform.model.entity

import com.google.gson.annotations.SerializedName
import java.io.Serializable

/**
 * 动态(朋友圈)实体数据模型，用于 Android 客户端。
 * 对应后端的 Moment MongoDB 文档。
 *
 * @property id 动态的唯一ID。
 * @property content 动态的文本内容。
 * @property images 图片URL列表的JSON字符串。客户端在需要时可以进一步解析此字符串。
 * @property authorId 作者的用户ID。
 * @property authorName 作者的用户名。
 * @property likeCount 动态的点赞数，默认为0。
 * @property comments 动态的评论列表，默认为空列表。
 * @property createdAt 动态的创建时间 (以ISO-8601格式字符串表示)。
 * @property updatedAt 动态的最后更新时间 (以ISO-8601格式字符串表示)。
 */
data class Moment(
    @SerializedName("id")
    val id: String,

    @SerializedName("content")
    val content: String?,

    @SerializedName("images")
    val images: String?,

    @SerializedName("authorId")
    val authorId: Long?,

    @SerializedName("authorName")
    val authorName: String?,

    @SerializedName("likeCount")
    val likeCount: Int? = 0,

    @SerializedName("comments")
    val comments: List<Comment>? = emptyList(),

    // 注意：后端字段为 snake_case，使用 SerializedName 进行映射
    @SerializedName("created_at")
    val createdAt: String?,

    @SerializedName("updated_at")
    val updatedAt: String?
) : Serializable